File: backend/app/schemas/alert.py
from __future__ import annotations

from typing import Optional
from pydantic import BaseModel


class AlertResponse(BaseModel):
    alert_id: str
    alert_type: str
    junction_id: Optional[str] = None
    junction_name: Optional[str] = None
    confidence: int
    message: str
    generated_at: str


class DismissResponse(BaseModel):
    alert_id: str
    status: str
    suppressed_until: str


class AlertPayload(BaseModel):
    alert_id: str
    severity: str  # "Watch", "Warning", "Critical"
    title: str
    description: str
    confidence: float
    created_at: str
    status: str  # "active", "acknowledged", "resolved"


class AcknowledgeRequest(BaseModel):
    alert_id: str


class AcknowledgeResponse(BaseModel):
    alert_id: str
    status: str


class ResolveRequest(BaseModel):
    alert_id: str


class ResolveResponse(BaseModel):
    alert_id: str
    status: str

File: frontend/types/alert.ts
export interface Alert {
  alert_id: string;
  alert_type: "escalation" | "hospital_corridor" | "officer_deficit" | "event_readiness" | "incident_spread";
  junction_id?: string;
  junction_name?: string;
  confidence: number;
  message: string;
  generated_at: string;
}

export interface DismissResponse {
  alert_id: string;
  status: string;
  suppressed_until: string;
}

export interface AlertPayload {
  alert_id: string;
  severity: "Watch" | "Warning" | "Critical";
  title: string;
  description: string;
  confidence: number;
  created_at: string;
  status: "active" | "acknowledged" | "resolved";
}

